tests/memory_checker/test_memory_checker.py

```diff
@@ -165,47 +290,109 @@ def consume_memory_and_restart_container(duthost, container_name, vm_workers, lo
-        return False
+    get_mem_usage_cmd = "docker stats --no-stream --format \{{\{{.MemUsage\}}\}} {}".format(container_name)
+    cmd_result = duthost.shell(get_mem_usage_cmd)
 
-    return True
+    exit_code = cmd_result["rc"]
+    pytest_assert(exit_code == 0, "Failed to get memory usage of '{}'!".format(container_name))
 
+    mem_info = cmd_result["stdout_lines"]
+    mem_usage = mem_info[0].split()[0]
+
+    return mem_usage
 
-def postcheck_critical_processes(duthost, container_name):
-    """Checks whether the critical processes are running after container was restarted.
+
+def consumes_memory_and_checks_monit(duthost, container_name, vm_workers, new_syntax_enabled):
+    """Invokes the 'stress' utility to consume memory more than the threshold asynchronously
+    and checks whether the container can be stopped and restarted. After container was restarted,
+    'stress' utility will be invoked again to consume memory and checks whether Monit was able to
+    restart this contianer with or without help of new syntax.
+    Loganalyzer is leveraged to check whether the log messages related to container stopped
+    and started were generated.
 
     Args:
         duthost: The AnsibleHost object of DuT.
         container_name: Name of container.
+        vm_workers: Number of workers which does the spinning on malloc()/free()
+          to consume memory.
+        new_syntax_enabled: Checks to make sure container will be restarted if it is set to be 
+          `True`.
 
     Returns:
         None.
     """
-    logger.info("Checking the running status of critical processes in '{}' container ..."
-                .format(container_name))
-    is_succeeded = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS, 0,
-                              check_critical_processes, duthost, container_name)
-    if not is_succeeded:
-        pytest.fail("Not all critical processes in '{}' container are running!"
-                    .format(container_name))
-    logger.info("All critical processes in '{}' container are running.".format(container_name))
+    expected_alerting_messages = []
+    expected_alerting_messages.append(".*restart_service.*Restarting service 'telemetry'.*")
+    expected_alerting_messages.append(".*Stopping Telemetry container.*")
+    expected_alerting_messages.append(".*Stopped Telemetry container.*")
+    expected_alerting_messages.append(".*Starting Telemetry container.*")
+    expected_alerting_messages.append(".*Started Telemetry container.*")
 
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="test_memory_checker")
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expected_alerting_messages)
+    marker = loganalyzer.init()
+
+    thread_pool = ThreadPool()
+    thread_pool.apply_async(consume_memory, (duthost, container_name, vm_workers))
+
+    logger.info("Sleep 130 seconds to wait for the alerting messages from syslog ...")
```

Comment:
>130 [](http://example.com/codeflow?start=23&length=3)

magic number

Reply:
Will use a named constant to represent this number.